### PR TITLE
Migrate old C++ unit tests to Python framework

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Others

### PR changes
Others

### Describe
We have few tests for fuse passes which are covered in `_mkldnn_fuse_pass_tester.cc`. These tests check only if graph has correct number of / type of vertices after fusion. It doesn't check accuracy. We think that current Python unit test framework is tailored for such tests, so we would like to rewrite obsolete tests to have full coverage. 

In this PR 3 tests are rewritten:
- `fc` + `activation` fuse pass,
- `softplus` + `activation` fuse pass,
- `fc` + `elementwise_add` fuse pass.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_act_fuse_pass.py
Comment (sfraczek) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_act_fuse_pass.py:35]: in new tests do you create gelu with approximate=False/True? I think not.

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_softplus_activation_fuse_pass.py
Comment (sfraczek) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_softplus_activation_fuse_pass.py:128]: Do you test approximate in new test

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_softplus_activation_fuse_pass.py
Comment (sfraczek) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_softplus_activation_fuse_pass.py:83]: swish had value 3 here for beta

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py
Comment (sfraczek) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py]: and here swish has different value range. Which is correct? In a comment of nn.py swish takes value 2.0
image

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py
Comment (sfraczek) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py]: Clip also takes different values now?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_act_fuse_pass.py
Comment (Silv3S) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_act_fuse_pass.py:35]: Thanks, I added approximate for `gelu` tests

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py
Comment (Silv3S) [reply python/paddle/fluid/tests/unittests/ir/inference/test_onednn_fc_activation_fuse_pass.py]: Swish `beta`, which is treated as `alpha` in oneDNN, can be any value in range (0, $\infty$). I will set random number in range [0.1, 10] as it's most popular.

**Review by sfraczek — DISMISSED** [review]
LGTM

**Review by jczaja — DISMISSED** [review]
LGTM

**Comment by Silv3S** [comment]
@kolinwei could you please review? We need approvals for deleting old unit tests, which were migrated to new UT framework

**Comment by Silv3S** [comment]
FC + activation fuse pass is tested already in `test_onednn_fc_activation_fuse_pass.py` which was already merged in https://github.com/PaddlePaddle/Paddle/pull/45183

**Comment by onecatcn** [comment]
@yeliang2258 Please help to review the PR. 
It seems being blocked by two CI checks, would you two discuss a solution to the issues? @Silv3S

**Comment by Silv3S** [comment]
> @yeliang2258 Please help to review the PR. It seems being blocked by two CI checks, would you two discuss a solution to the issues? @Silv3S

This PR is blocked only by QA approval, because some unit tests are removed / migrated to new UT framework. We decided to rewrite old unit tests as they did not catch all bugs (accuracy difference in softplus+activation is good example).

Error message:

> It is forbidden to disable or delete the unit-test.
> If you must delete it temporarily, please add it to[https://github.com/PaddlePaddle/Paddle/wiki/Temporarily-disabled-Unit-Test].
> Then you must have one RD (kolinwei(recommended), chalsliu, XieYunshen or zhouwei25) approval for the deletion of unit-test. 
> If you have any problems about deleting unit-test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/Deleting-unit-test-is-forbidden].

**Review by chenwhql — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]
LGTM 
单测调整

**Review by jczaja — DISMISSED** [review]

**Review by chenwhql — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]

**Comment by Silv3S** [comment]
@jczaja please merge. PR-CI-Kunlun-KP-Build is again broken for any PR, but fortunately this test is not required

**Pull request merged**
